### ISSUE TITLE
Regression: unable to find linter when using pyenv

### ISSUE THREAD

**Issue by ssbarnea** [ansible/ansible-language-server#562]
### Summary

![](https://sbarnea.com/ss/Screen-Shot-2023-04-11-16-06-16.40.png)

When installing scripts using pyenv, these are shell scripts and the command that we seen as failing to run makes no sense, extension tried to use python3 to executed the linter bash scripts!!

### Extension version

v1.3.50

### VS Code version

1.77.1

### Ansible Version

```shell
-
```

### OS / Environment

macos with pyenv python

**Comment by ssbarnea** [comment]
The bug is caused by the withInterpreter function implementation which makes some wrong assumptions, https://github.com/ansible/ansible-language-server/blob/959ad907848971df2bd69ac129ca0da41002a2e7/src/utils/misc.ts#LL44C17-L44C32

To reproduce:
- use pyenv
- configure python interpreter path
- configure ansible-lint to point to pyenv installed ansible-lint, see example below:

```
$ which -a ansible-lint
/Users/ssbarnea/.pyenv/shims/ansible-lint

$ cat /Users/ssbarnea/.pyenv/shims/ansible-lint | head -n 1
#!/usr/bin/env bash
```

As you can see, for the pyenv installed shims are not python scripts, they are bash scripts, so our assumption that we can use python to call them was a mistake.

Conclusion: when someone is configuring a path to an executable, we should never try to use a specific application to run them, either bash or python. Instead we should just execute them.

This means that configuring python interpreter path does not override a customly configured ansible-lint path. We should document this.

The only case where it is ok for use to use python to call the linter, or something else, is if the user did not configure a custom path for that tool.

**Comment by ssbarnea** [comment]
I made some tests and verified that the bad review we had on marketplace about not being able to use ansible-lint installed from conda proved to be caused by the same bug.

```shell
$ cat /opt/homebrew/Caskroom/miniconda/base/bin/ansible-lint
#!/opt/homebrew/Caskroom/miniconda/base/bin/python3.10
# -*- coding: utf-8 -*-
import re
import sys

from ansiblelint.__main__ import main

if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
(base)
```

If you look at this, it a shell script with a custom shebang like that points to the conda python.

If we do use python to call it it will fail to either run, or it will use another incorrect version of ansible.

In both cases the root cause is the same: failure to allow the shell to choose the executable. 

Basically when we configure a custom python interpreter path, that should never have an effect on commands that are running executables/scripts. That custom python interpreter would be safe to call only Python modules, not files marked as executable.
